src/forms/EditEquipmentListingGeneralForm/EditEquipmentListingGeneralForm.js

```diff
@@ -109,14 +123,24 @@ const EditEquipmentListingGeneralFormComponent = props => (
       const submitDisabled = invalid || disabled || submitInProgress;
 
       const options = findOptionsForSelectFilter('equipment', filterConfig);
-      const handleManufactureOnChange = event => {
+
+      const callManufactureValidate = (value) => {
+        return validateYear(yearErrorMessage, value)
+      }
+
+      const callHourValidate = (value) => {
+        return validateHour(hourErrorMessage, value)
+      }
+
+      const handleNumberFieldOnChange = (fieldName, event) => {
```

Comment:
You should take a look in the `validators.js` file to know how to write a validation using currying function. And all validator should be declared in validators file.

Reply:
I have improved this, can you take a look?
https://github.com/trinh-le-jh/ftw-daily/commit/4be4107f8f76cf4e36579497e508885bf56e2612